homeassistant/components/proxmoxve/binary_sensor.py

```diff
@@ -74,7 +84,9 @@ def device_state_attributes(self):
             "node": self._item_node,
             "vmid": self._item_id,
             "vmname": self._vmname,
-            "type": self._item_type.name,
+            "type": self._item_type,
+            "cpu": self._cpus,
+            "memory": self._memory / (1024 * 1024),
```

Comment:
These are already sensors in this PR, why create duplicates?

Reply:
the intention is, that you have an overview about one single guest instance like
![image](https://user-images.githubusercontent.com/35783820/98411565-913f3300-2076-11eb-8c91-33806350e041.png)


But meanwhile it was thinking about to refactor the whole integration, so that

- each guest instance is represented by an device which has multiple sensors available
- config flow is used instead of yaml configuration
- services for start/stop/reboot of guest instances are available
- all available nodes and guest instances can be discovered, instead of define each of them manually in config
- filters for nodes and guest instances for discovery can be set via config flow

But I´m not sure if this PR should be extended by or I have to create a new PR for above features?

Reply:
We try to avoid attributes in general. It pollutes the state machine and triggers more updates than needed (in general).

We prefer separate sensors. How it is displayed is a frontend problem, and can be formatted by a user in Lovelace the way they want (using multiple entities).

Reply:
@frenck what du you think about the idea of refactoring and if this PR should be extended by or a should I create a new PR for the new planned features?

Reply:
I would make smaller steps and avoid creating big PRs

Reply:
ok, than I will create a after this one a new PR for the braking change to switch to one device per guest instance with sensors and afterwards, I will create different PRs per new feature, because they partially depends on each other.